internal/api/resolver_mutation_studio.go

```diff
@@ -28,190 +24,284 @@ func (r *mutationResolver) getStudio(ctx context.Context, id int) (ret *models.S
 }
 
 func (r *mutationResolver) StudioCreate(ctx context.Context, input StudioCreateInput) (*models.Studio, error) {
-	// generate checksum from studio name rather than image
-	checksum := md5.FromString(input.Name)
-
-	var imageData []byte
+	var studioID *int
+	var dbInput models.StudioDBInput
 	var err error
+	var parentTranslator changesetTranslator
+	runParentCreateHook := false
+	runParentUpdateHook := false
+
+	if input.Parent != nil {
+		if input.ParentID == nil {
+			// The parent needs to be created
+			dbInput.ParentCreate, err = studioFromStudioCreateInput(ctx, *input.Parent)
+			runParentCreateHook = true
+		} else {
+			parentTranslator = changesetTranslator{
+				inputMap: getNamedUpdateInputMap(ctx, updateInputField+".parent"),
+			}
 
-	// Process the base 64 encoded image string
-	if input.Image != nil {
-		imageData, err = utils.ProcessImageInput(ctx, *input.Image)
+			// The parent studio matched an existing one and the user has chosen in the UI to link and/or update it
+			dbInput.ParentUpdate, err = studioPartialFromStudioCreateInput(ctx, *input.Parent, input.ParentID, parentTranslator)
+			runParentUpdateHook = true
+		}
 		if err != nil {
 			return nil, err
 		}
 	}
 
+	dbInput.StudioCreate, err = studioFromStudioCreateInput(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+
+	// Start the transaction and save the studio
+	if err := r.withTxn(ctx, func(ctx context.Context) error {
+		qb := r.repository.Studio
+		studioID, err = qb.Create(ctx, dbInput)
+		if err != nil {
+			return err
+		}
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	newStudio, err := r.getStudio(ctx, *studioID)
+	if err != nil {
+		return nil, fmt.Errorf("finding after create: %w", err)
+	}
+
+	if runParentCreateHook {
+		r.hookExecutor.ExecutePostHooks(ctx, *newStudio.ParentID, plugin.StudioCreatePost, input, nil)
+	} else if runParentUpdateHook {
+		r.hookExecutor.ExecutePostHooks(ctx, *newStudio.ParentID, plugin.StudioUpdatePost, input, parentTranslator.getFields())
+	}
+	r.hookExecutor.ExecutePostHooks(ctx, *studioID, plugin.StudioCreatePost, input, nil)
+
+	return newStudio, nil
+}
+
+func studioFromStudioCreateInput(ctx context.Context, input StudioCreateInput) (*models.Studio, error) {
 	// Populate a new studio from the input
-	currentTime := time.Now()
 	newStudio := models.Studio{
-		Checksum:  checksum,
```

Comment:
I moved checksum and the date fields into model_studio.go NewStudio and NewStudioPartial to avoid duplicating it everywhere, but looking now that doesn't stop anyone from just making a new object manually without using those methods and missing out on the fields. I will review this further.

Reply:
Confirmed CreatedAt/UpdatedAt are enforced at the sqlite level.